pkg/controller/authentication/routes.go

```diff
@@ -458,6 +459,9 @@ func (r *ReconcileAuthentication) newRoute(instance *operatorv1alpha1.Authentica
 
 	weight := int32(100)
 
+	commonLabel := map[string]string{"app": "im"}
+	routeLabels := pkgCommon.MergeMap(commonLabel, instance.Spec.Labels)
```

Comment:
If the intent is for the `commonLabel` labels to be preserved, then the arguments should be reordered.

Reply:
i have  swapped it in the source function(MergeMap) body